CliFx.Tests/SuggestDirectiveSpecs.cs

```diff
@@ -0,0 +1,288 @@
+        }
+
+        [Fact]
+        public async Task Suggest_directive_is_disabled_by_default()
+        {
+            // Arrange
+            var application = TestApplicationFactory(_cmdCommandCs)
+                .Build();
+
+            // Act
+            var exitCode = await application.RunAsync(
+                new[] { "[suggest]", "clifx.exe", "c" }
+            );
+
+            // Assert
+            exitCode.Should().Be(1);
+        }
+
+        [Theory]
+        [InlineData("supply all commands if nothing supplied",
+                        "clifx.exe", 0, new[] { "cmd", "cmd02", "parent", "parent list" })]
+        [InlineData("supply all commands that 'start with' argument",
+                        "clifx.exe c", 0, new[] { "cmd", "cmd02" })]
+        [InlineData("supply command options if match found, regardles of other partial matches (no options defined)",
+                        "clifx.exe cmd", 0, new string[] { })]
+        [InlineData("supply nothing if no commands 'starts with' argument",
+                        "clifx.exe m", 0, new string[] { })]
+        [InlineData("supply completions of partial child commands",
+                        "clifx.exe parent l", 0, new[] { "list" })]
+        [InlineData("supply all commands that 'start with' argument, allowing for cursor position",
+                        "clifx.exe cmd", -2, new[] { "cmd", "cmd02" })]
+        public async Task Suggest_directive_suggests_commands_by_environment_variables(string usecase, string variableContents, int cursorOffset, string[] expected)
+        {
+            // Arrange
+            var application = TestApplicationFactory(_cmdCommandCs, _cmd2CommandCs, _parentCommandCs, _childCommandCs)
+                .AllowSuggestMode()
+                .Build();
+
+            // Act
+            var exitCode = await application.RunAsync(
+                new[] { "[suggest]", "--envvar", "CLIFX-{GUID}", "--cursor", (variableContents.Length + cursorOffset).ToString() },
+                new Dictionary<string, string>()
+                {
+                    ["CLIFX-{GUID}"] = variableContents
+                }
+            );
+
+            var stdOut = FakeConsole.ReadOutputString();
+
+            // Assert
+            exitCode.Should().Be(0);
+
+            stdOut.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries)
+                  .Should().BeEquivalentTo(expected, usecase);
+        }
+
+        [Theory]
+        [InlineData("supply all commands that match partially",
+                        new[] { "[suggest]", "clifx.exe", "c" }, new[] { "cmd", "cmd02" })]
+        [InlineData("supply command options if match found, regardles of other partial matches (no options defined)",
+                        new[] { "[suggest]", "clifx.exe", "cmd" }, new string[] { })]
+        public async Task Suggest_directive_suggests_commands_by_command_line_only(string usecase, string[] commandLine, string[] expected)
```

Comment:
Do I understand correctly that `[suggest]` works with both arguments provided via environment variable and through command line? Do we need both?